### DISCUSSION TITLE
Why is introduction item shown in each group in Sidebar?

### DISCUSSION THREAD

**Discussion by eniskraasniqi1** [PaloAltoNetworks/docusaurus-openapi-docs#1004] [Q&A]
### Description

Hi! 👋
Is there a way to remove the Introduction item from appearing in each group within the sidebar and instead display it either inside a standalone group or only within one specific group? This issue only arises when generating sidebar using grouping by tagGroup.

```
sidebarOptions: {
	groupPathsBy: 'tagGroup',
},
```

### With `tagGroup`

![SCR-20241025-l2y](https://github.com/user-attachments/assets/3a93ca98-fd8a-47c9-a363-ce8e45e5975c)

### With `tag`

Whereas when you group by tag, its only shown once in the top.

```
sidebarOptions: {
	groupPathsBy: 'tag',
}
```

![SCR-20241025-l7r](https://github.com/user-attachments/assets/af84d48b-bb94-4394-9b93-d7b991fdfb44)

### Examples

- https://docusaurus-openapi.tryingpan.dev/restaurant/restaurant-example (All introduction items are expanded).
- https://codesandbox.io/p/devbox/3tnxvv

**Comment by aszenz** [comment]
I'm facing the exact same issue, not sure if we are doing something wrong or its intended behaviour

**Reply by aszenz** [reply]
@eniskraasniqi1 I suggest converting this discussion into an issue

**Reply by eniskraasniqi1** [reply]
Hi @aszenz!
Still not sure if its intended behaviour, so I thought of waiting a bit more until an answer is provided. 
At the same time, I went with a custom solution as a temporary fix, I'll share it with you also just in case it helps:

```
// scripts/filterAutogeneratedSidebar.js

const mySubItemId = ' id: "api/my-api"'; // you can find this in the `sidebar.ts` items, its always the first in each category.

export function filterAutogeneratedSidebar(autogeneratedSidebar) {
  return (autogeneratedSidebar || []).map((item) => {
    if (item.type === 'category') {
      return {
        ...item,
        items: item.items.filter((subItem) => {
          if (subItem.type === 'doc' && subItem.id.includes(mySubItemId)) {
            return false;
          }

          return true;
        }),
      };
    }
    return item;
  });
}
```

```
// sidebars.ts

import autogeneratedSidebar from './docs/api/sidebar';
import { filterAutogeneratedSidebar } from './scripts/filterAutogeneratedSidebar.js';

const filteredSidebar = filterAutogeneratedSidebar(autogeneratedSidebar);
const sidebars: SidebarsConfig = {
  mySidebar: [
    {
      type: 'category',
      label: 'My sidebar',
      link: {
        type: 'generated-index',
        title: 'API Documentation',
        slug: '/api',
      },
      items: filteredSidebar, // you can pass the fitlered items here
    },
  ],
};
```
